benchmarks/speedtest/src/opnsense/mvc/app/controllers/OPNsense/Speedtest/Api/ServiceController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * Copyright 2021 Miha Kralj
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+namespace OPNsense\Speedtest\Api;
+
+use OPNsense\Base\ApiControllerBase;
+use OPNsense\Core\Backend;
+
+class ServiceController extends ApiControllerBase
+{
+
+    public function listAction()
+    {
+        $backend = new Backend();
+        $response = $backend->configdRun("speedtest serverlist");
+        return array("response" => $response);
+    }
+
+    public function testAction($serverid = 0)
+    {
+        $backend = new Backend();
+        $response = trim($backend->configdRun("speedtest run ${serverid}"));
```

Comment:
you can pass parameters using  `->configdpRun()` e.g.

```
(new Backend())->configdpRun("speedtest run", [$serverid]);
```

This helps to make the distinction between commands and parameters more explicit and escapes parameters in the process.

Reply:
Do you want me to refactor the controller for this release or schedule the changes for the next update? This jello works at the moment, I will create a new fork and start on optimization and refactoring.